redis/src/types.rs

```diff
@@ -810,38 +832,53 @@ impl RedisError {
         Self { repr }
     }
 
-    // TODO: In addition to/instead of returning a bool here, consider a method
-    // that returns an enum with more detail about _how_ to retry errors, e.g.,
-    // `RetryImmediately`, `WaitAndRetry`, etc.
-    #[cfg(feature = "cluster")] // Used to avoid "unused method" warning
-    pub(crate) fn is_retryable(&self) -> bool {
+    pub(crate) fn retry_method(&self) -> RetryMethod {
         match self.kind() {
-            ErrorKind::BusyLoadingError => true,
-            ErrorKind::Moved => true,
-            ErrorKind::Ask => true,
-            ErrorKind::TryAgain => true,
-            ErrorKind::MasterDown => true,
-            ErrorKind::IoError => true,
-            ErrorKind::ReadOnly => true,
-            ErrorKind::ClusterDown => true,
-            ErrorKind::MasterNameNotFoundBySentinel => true,
-            ErrorKind::NoValidReplicasFoundBySentinel => true,
-
-            ErrorKind::ExtensionError => false,
-            ErrorKind::ExecAbortError => false,
-            ErrorKind::ResponseError => false,
-            ErrorKind::AuthenticationFailed => false,
-            ErrorKind::TypeError => false,
-            ErrorKind::NoScriptError => false,
-            ErrorKind::InvalidClientConfig => false,
-            ErrorKind::CrossSlot => false,
-            ErrorKind::ClientError => false,
-            ErrorKind::EmptySentinelList => false,
-            ErrorKind::NotBusy => false,
+            ErrorKind::Moved => RetryMethod::MovedRedirect,
+            ErrorKind::Ask => RetryMethod::AskRedirect,
+
+            ErrorKind::TryAgain => RetryMethod::WaitAndRetry,
+            ErrorKind::MasterDown => RetryMethod::WaitAndRetry,
+            ErrorKind::ClusterDown => RetryMethod::WaitAndRetry,
+            ErrorKind::BusyLoadingError => RetryMethod::WaitAndRetry,
```

Comment:
But why do we need to wait on a loading error - we can immediately retry it on a different replica/on the primary. If the primary raise the exception and we don't have other replicas/can't redirect to replicas we might want to wait and retry, but most times it won't be the case

Reply:
Once the loading error PR lands, there will be another behavior, for loading errors, so it doesn't really matter what behavior we give loading errors.